### PR TITLE
Switch from `RevSort.TOPO` to `RevSort.COMMIT_TIME_DESC` to improve performance on large repos

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java [hunk 1]
```diff
@@ -572,7 +576,12 @@ public GitCoreRepositoryState deriveRepositoryState() {
   @Override
   public Stream<IGitCoreCommit> ancestorsOf(IGitCoreCommit commitInclusive) throws GitCoreException {
     RevWalk walk = new RevWalk(jgitRepoForMainGitDir);
-    walk.sort(RevSort.TOPO);
```
Comment (PawelLipski) [inline gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java:575]: _he found one simple trick_

File: .circleci/config.yml [hunk 2]
```diff
@@ -232,15 +232,15 @@ jobs:
                 # language=sh
                 command: |
                   source scripts/utils.sh
-                  corresponding_milestone_title=v$(extract_version_from_current_wd)
-                  tag=$corresponding_milestone_title
+                  milestone_title=v$(extract_version_from_current_wd)
```
Comment (PawelLipski) [inline .circleci/config.yml:235]: Unrel

File: scripts/close-milestone [hunk 3]
```diff
@@ -1,21 +1,16 @@
 #!/usr/bin/env bash
 
-set -x -e -o pipefail -u
+set -e -o pipefail -u -x
 
 milestone_title=$1
 
-milestones_json_array=$(gh api -H "Accept: application/vnd.github+json" /repos/VirtusLab/git-machete-intellij-plugin/milestones)
+milestones_json_array=$(gh api repos/VirtusLab/git-machete-intellij-plugin/milestones)
```
Comment (PawelLipski) [inline scripts/close-milestone:7]: More unrelated tidy-up

File: gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java [hunk 4]
```diff
@@ -533,7 +533,11 @@ public List<IGitCoreCommit> deriveCommitRange(IGitCoreCommit fromInclusive, IGit
     LOG.debug(() -> "Entering: fromInclusive = '${fromInclusive}', untilExclusive = '${untilExclusive}'");
 
     return withRevWalk(walk -> {
-      walk.sort(RevSort.TOPO);
+      // Note that `RevSort.COMMIT_TIME_DESC` is compatible with git-machete CLI,
+      // which relies on vanilla `git log` under the hood,
+      // which by default shows commits in reverse chronological order (https://git-scm.com/docs/git-log#_commit_ordering).
+      // In this case (unlike with `ancestorsOf`), apparently there is no significant effect on performance.
```
Comment (mkondratek) [inline gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java:539]: 😕

File: gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java [hunk 5]
```diff
@@ -572,7 +576,12 @@ public GitCoreRepositoryState deriveRepositoryState() {
   @Override
   public Stream<IGitCoreCommit> ancestorsOf(IGitCoreCommit commitInclusive) throws GitCoreException {
     RevWalk walk = new RevWalk(jgitRepoForMainGitDir);
-    walk.sort(RevSort.TOPO);
+    // Note that `RevSort.COMMIT_TIME_DESC` is both:
+    // * compatible with git-machete CLI, which relies on vanilla `git log` under the hood,
+    //   which by default shows commits in reverse chronological order (https://git-scm.com/docs/git-log#_commit_ordering),
+    // * significantly faster than `RevSort.TOPO` on repos with large histories (100,000's of commits),
```
Comment (mkondratek) [inline gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java:582]: 🚀

File: src/test/java/com/virtuslab/archunit/ForbiddenFieldsTestSuite.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+public class ForbiddenFieldsTestSuite extends BaseArchUnitTestSuite {
+
+  @Test
+  public void no_classes_should_access_RevSort_TOPO() {
+    noClasses()
+        .should()
+        .accessField(org.eclipse.jgit.revwalk.RevSort.class, "TOPO")
+        .orShould()
+        .accessField(org.eclipse.jgit.revwalk.RevSort.class, "TOPO_KEEP_BRANCH_TOGETHER")
+        .because("as of JGit 6.4.0, these flags lead to performance problems on large repos (100,000s of commits); " +
```
Comment (mkondratek) [inline src/test/java/com/virtuslab/archunit/ForbiddenFieldsTestSuite.java:16]: :heart:

**Review by mkondratek — APPROVED** [review]

**Pull request merged**